Doc/library/os.rst

```diff
@@ -3780,6 +3780,174 @@ features:
+   :func:`~select.select`, and :func:`~select.poll`.
+
+   .. seealso:: The :manpage:`timerfd_create(2)` man page.
+
+   *clockid* must be a valid :ref:`clock ID <time-clock-id-constants>`,
+   as defined in the :py:mod:`time` module.
+
+   The file descriptor's behaviour can be modified by specifying a *flags* value.
+   Any of the following variables may used, combined using bitwise OR
+   (the ``|`` operator):
+
+   - :const:`TFD_NONBLOCK`
+   - :const:`TFD_CLOEXEC`
+
+   If the timer file descriptor's counter is zero and
+   :const:`TFD_NONBLOCK` is not set as a flag, :func:`read` blocks.
+
+   The following example shows how to use a timer file descriptor
+   to execute a function twice a second:
+
+   .. code:: python
+
+      import os, time
+
+      fd = os.timerfd_create(time.CLOCK_REALTIME, 0)
+      interval = 0.5  # Set the timer interval to 0.5 seconds
+      value = 1  # Start the timer in 1 second
+      os.timerfd_settime(fd, 0, interval, value)  # Start the timer
+
+      try:
+          while:
+              fd.read(8)  # Wait for the timer to expire
+              do_work()
+      finally:
+          os.close(fd)
+
+   Units may be specified in nanoseconds for greater precision.
+   Repeating the previous example:
+
+   .. code:: python
+
+      import os, time
+
+      fd = os.timerfd_create(time.CLOCK_REALTIME, 0)
+      interval = 10**9 // 2  # Set the timer interval to 0.5 seconds
+      value = 10**9  # Start the timer in 1 second
+      os.timerfd_settime_ns(fd, 0, interval, value)  # Start the timer
+
+      try:
+          while:
+              fd.read(8)  # Wait for the timer to expire
+              do_work()
+      finally:
+          os.close(fd)
+
+   .. availability:: Linux >= 2.6.27 with glibc >= 2.8
+
+   .. versionadded:: 3.13
+
+
+.. function:: timerfd_settime(fd, flags, interval=0.0, value=0.0, /)
+
+   Alter a timer file descriptor's internal timer.
+
+   .. seealso::  The :manpage:`timerfd_settime(2)` man page.
+
+   *fd* must be a valid timer file descriptor.
+
+   The timer's behaviour can be modified by specifying a *flags* value.
+   Any of the following variables may used, combined using bitwise OR
+   (the ``|`` operator):
+
+   - :const:`TFD_TIMER_ABSTIME`
+   - :const:`TFD_TIMER_CANCEL_ON_SET`
+
+   The timer's interval is set by the *interval* :py:class:`float`.
+   If *interval* is zero, the timer only fires once, on the initial expiration.
+   If *interval* is greater than zero, the timer fires every time *interval*
+   seconds have elapsed since the previous expiration.
+
+   The timer is disabled by setting *value* to zero (``0``).
+   If *value* is set to a non-zero value, the timer is enabled.
+   By default timer will fire when *value* seconds have elapsed.
+   However, if the :const:`TFD_TIMER_ABSTIME` flag is set,
+   the timer will fire when the timer's clock
+   (set by *clockid* in :func:`timerfd_create`) reaches *value* seconds.
```

Comment:
Is there any way to query the internal clock given an arbitrary timerfd?

Reply:
@AA-Turner 
I think there is no such operation.

Reply:
Should we add one?

Reply:
@AA-Turner 
You mean we should save the internal clock inside and return the saved value if requested?